internal/controller/monovertexrollout_controller.go

```diff
@@ -336,10 +337,10 @@ func (r *MonoVertexRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Group:   common.NumaflowAPIGroup,
 		Version: common.NumaflowAPIVersion,
 	})
-	if err := controller.Watch(source.Kind(mgr.GetCache(), monoVertexUns),
-		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &apiv1.MonoVertexRollout{}, handler.OnlyControllerOwner()),
-		predicate.ResourceVersionChangedPredicate{}); err != nil {
-		return err
+	if err := controller.Watch(source.Kind(mgr.GetCache(), monoVertexUns,
+		handler.TypedEnqueueRequestForOwner[*unstructured.Unstructured](mgr.GetScheme(), mgr.GetRESTMapper(),
+			&apiv1.MonoVertexRollout{}, handler.OnlyControllerOwner()), predicate.TypedResourceVersionChangedPredicate[*unstructured.Unstructured]{})); err != nil {
+		return fmt.Errorf("failed to watch MonoVertices: %w", err)
```

Comment:
Good job figuring out how to make this work for `*Unstructured` type - that's too bad that this stuff wasn't backward compatible as is